scipy/linalg/special_matrices.py

```diff
@@ -236,13 +235,18 @@ def circulant(c):
            [2, 1, 3],
            [3, 2, 1]])
 
+    References
+    ----------
+    .. [1] Stack Overflow, "Create a matrix from a vector where each row is a
+           shifted version of the vector", https://stackoverflow.com/a/43735362
+
     """
     c = np.asarray(c).ravel()
-    a, b = np.ogrid[0:len(c), 0:-len(c):-1]
-    indx = a + b
-    # `indx` is a 2D array of indices into `c`, arranged so that `c[indx]` is
-    # the circulant matrix.
-    return c[indx]
+    # Form an extended array that could be strided to give circulant version
+    c_ext = np.concatenate((c[::-1], c[:0:-1]))
+    L = len(c)
+    n = c_ext.strides[0]
+    return as_strided(c_ext[L-1:], (L, L), (-n, n)).copy()
```

Comment:
Use the explicit keyword arguments, like you did in `hankel` and `toeplitz`:

    return as_strided(c_ext[L-1:], shape=(L, L), strides=(-n, n)).copy()